include/libopencm3/stm32/mp1/memorymap.h

```diff
@@ -0,0 +1,194 @@
+#define LPTIM1_BASE                     (PERIPH_BASE_APB1 + 0x09000)
+#define WWDG_BASE                       (PERIPH_BASE_APB1 + 0x0a000)
+#define SPI2_BASE                       (PERIPH_BASE_APB1 + 0x0b000)
+#define SPI3_BASE                       (PERIPH_BASE_APB1 + 0x0c000)
+#define SPDIFRX_BASE                    (PERIPH_BASE_APB1 + 0x0d000)
+#define USART2_BASE                     (PERIPH_BASE_APB1 + 0x0e000)
+#define USART3_BASE                     (PERIPH_BASE_APB1 + 0x0f000)
+#define UART4_BASE                      (PERIPH_BASE_APB1 + 0x10000)
+#define UART5_BASE                      (PERIPH_BASE_APB1 + 0x11000)
+#define I2C1_BASE                       (PERIPH_BASE_APB1 + 0x12000)
+#define I2C2_BASE                       (PERIPH_BASE_APB1 + 0x13000)
+#define I2C3_BASE                       (PERIPH_BASE_APB1 + 0x14000)
+#define I2C5_BASE                       (PERIPH_BASE_APB1 + 0x15000)
+#define CEC_BASE                        (PERIPH_BASE_APB1 + 0x16000)
+#define DAC_BASE                        (PERIPH_BASE_APB1 + 0x17000)
+#define UART7_BASE                      (PERIPH_BASE_APB1 + 0x18000)
+#define UART8_BASE                      (PERIPH_BASE_APB1 + 0x19000)
+#define MDIOS_BASE                      (PERIPH_BASE_APB1 + 0x1c000)
+
+/* APB2 */
+#define TIM1_BASE                       (PERIPH_BASE_APB2 + 0x00000)
+#define TIM8_BASE                       (PERIPH_BASE_APB2 + 0x01000)
+#define USART6_BASE                     (PERIPH_BASE_APB2 + 0x03000)
+#define SPI1_BASE                       (PERIPH_BASE_APB2 + 0x04000)
+#define SPI4_BASE                       (PERIPH_BASE_APB2 + 0x05000)
+#define TIM15_BASE                      (PERIPH_BASE_APB2 + 0x06000)
+#define TIM16_BASE                      (PERIPH_BASE_APB2 + 0x07000)
+#define TIM17_BASE                      (PERIPH_BASE_APB2 + 0x08000)
+#define SPI5_BASE                       (PERIPH_BASE_APB2 + 0x09000)
+#define SAI1_BASE                       (PERIPH_BASE_APB2 + 0x0a000)
+#define SAI2_BASE                       (PERIPH_BASE_APB2 + 0x0b000)
+#define SAI3_BASE                       (PERIPH_BASE_APB2 + 0x0c000)
+#define DFSDM1_BASE                     (PERIPH_BASE_APB2 + 0x0d000)
+#define FDCAN1_BASE                     (PERIPH_BASE_APB2 + 0x0e000)
+#define FDCAN2_BASE                     (PERIPH_BASE_APB2 + 0x0f000)
+#define CAN_CCU_BASE                    (PERIPH_BASE_APB2 + 0x10000)
+#define CAN_SRAM_BASE                   (PERIPH_BASE_APB2 + 0x11000)
+
+/* AHB2 */
+#define DMA1_BASE                       (PERIPH_BASE_AHB2 + 0x00000)
+#define DMA2_BASE                       (PERIPH_BASE_AHB2 + 0x01000)
+#define DMAMUX_BASE                     (PERIPH_BASE_AHB2 + 0x02000)
+#define ADC12_BASE                      (PERIPH_BASE_AHB2 + 0x03000)
+#define ADC1_BASE                       (ADC12_BASE + 0x000)
+#define ADC2_BASE                       (ADC12_BASE + 0x100)
+#define ADC_CMN_BASE                    (ADC12_BASE + 0x300)
```

Comment:
this looks suspcious.  we already have some ofthis style adc, and it's defined differently.  I'll try and find the right examples to copy, it's _at least_ in include/libopencm3/stm32/common/adc_common_v1_multi.h